File: DSAinJava/BinaryTree/Maximumdethoftree.java
/**
 * leetcode problem no :(104)
 * 
 * leetcode solution link (https://leetcode.com/problems/maximum-depth-of-binary-tree/)
 * 
 */

 /**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int f(TreeNode root){
        if(root == null){
            return 0;
        }
        // if root is not null then ,
       int result = 1+ Math.max(f(root.left) , f(root.right));
         return result;
    }
    public int maxDepth(TreeNode root) {

       return  f(root);
        
    }
}

File: DSAjavascript/BinarryTree/Maximumdethoftree.js
/**
 * leetcode problem no :(104)
 * 
 * leetcode solution link (https://leetcode.com/problems/maximum-depth-of-binary-tree/)
 * 
 */

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
function f(root){
    if(root == null){
        return 0;
    }

   let result = 1 + Math.max(f(root.left) , f(root.right));
   return result;
 }
var maxDepth = function(root) {
    return f(root);
    
};